core/google/api/core/future/polling.py

```diff
@@ -69,29 +71,21 @@ def _blocking_poll(self, timeout=None):
         if self._result_set:
             return
 
-        retry_on = tenacity.retry_if_result(
-            functools.partial(operator.is_not, True))
-        # Use exponential backoff with jitter.
-        wait_on = (
-            tenacity.wait_exponential(multiplier=1, max=10) +
-            tenacity.wait_random(0, 1))
-
-        if timeout is None:
-            retry = tenacity.retry(retry=retry_on, wait=wait_on)
-        else:
-            retry = tenacity.retry(
-                retry=retry_on,
-                wait=wait_on,
-                stop=tenacity.stop_after_delay(timeout))
+        def done_or_raise():
+            """Checks if the future is done and raises if it's not."""
+            if not self.done():
+                raise _OperationNotComplete()
+
+        retry_ = retry.Retry(
+            predicate=retry.if_exception_type(_OperationNotComplete),
+            deadline=timeout)
 
         try:
-            retry(self.done)()
-        except tenacity.RetryError as exc:
-            six.raise_from(
-                concurrent.futures.TimeoutError(
-                    'Operation did not complete within the designated '
-                    'timeout.'),
-                exc)
+            retry_(done_or_raise)()
+        except exceptions.RetryError:
+            raise concurrent.futures.TimeoutError(
+                'Operation did not complete within the designated '
+                'timeout.')
```

Comment:
the `_OperationNotComplete` exception doesn't provide any useful information or context.